### PR TITLE
Additional Prefrences dialog

### PR DESCRIPTION
**Pull request by peterooch**
Added a dialog that enables a selection of text editor and view other than notepad,
currently works only if manually specifiying a path to the editor.
The "..." buttons throws an access error.

Saves the path in the INI file

Includes fallback to notepad if something bad happens

Made by inspiration from Issue #115 

(also uses wcs* functions instead of the tcs* equivalents)

### CODE REVIEW

File: src/res.h [hunk 1]
```diff
@@ -144,11 +144,13 @@
 
 #define IDM_TOOLBARCUST     512
 
+#define IDM_EDITOR          513
```
Comment (craigwi) [inline src/res.h]: No need to renumber existing entries here.  Just add later in the file.

File: src/wfcomman.c [hunk 2]
```diff
@@ -690,31 +690,24 @@ OpenOrEditSelection(HWND hwndActive, BOOL fEdit)
...
-            if (!PathFileExists(szToRun))
-            {
-                cchEnv = 0;
-            }
-          }
+      {   
 
-          if (cchEnv == 0)
-          {
-              // NOTE: assume system directory and "\\notepad.exe" never exceed MAXPATHLEN
```
Comment (craigwi) [inline src/wfcomman.c]: Carry forward this comment.

File: src/wfdlgs.c [hunk 3]
```diff
@@ -698,6 +698,72 @@ ConfirmDlgProc(HWND hDlg, UINT wMsg, WPARAM wParam, LPARAM lParam)
...
+            break;
+        
+        case WM_COMMAND:
+            switch (GET_WM_COMMAND_ID(wParam, lParam))
+            {
+                case IDD_HELP:
+                    goto DoHelp;
+
+                case IDC_EDITOR:
+                    GetOpenFileName(&ofn);
```
Comment (craigwi) [inline src/wfdlgs.c]: This should use the LoadComdlg() pattern elsewhere in the code.

File: src/winfile.h [hunk 4]
```diff
@@ -962,7 +963,7 @@ Extern DWORD (*lpfnGetOpenFileNameW)(LPOPENFILENAMEW);
 #define COMDLG_GetOpenFileNameW           "GetOpenFileNameW"
 
 #define ChooseFontW         (*lpfnChooseFontW)
-#define GetOpenFileNameW    (*lpfnGetOpenFileNameW)
+//#define GetOpenFileNameW    (*lpfnGetOpenFileNameW)
```
Comment (craigwi) [inline src/winfile.h]: You should not need to remove this.

**Comment by peterooch** [comment]
prefdlg_show 2

Added more stuff... is it too much feature creeping for a PR or it would be a welcome addition/change?

**Comment by thecatkitty** [comment]
@peterooch, I know that it's not an answer for your question, but wouldn't it make more sense if the string "Chinese (Simplified)" were written in Chinese?

**Comment by peterooch** [comment]
@thecatkitty, Its in a placeholder string table, just to see if it works, the current branch doesn't have the a9fe81f code even, but i had to have something to differentiate from English. I also would prefer "Local Name (English Name)" or the other way around for the list values.

**Comment by craigwi** [comment]
I like the direction.  We should move the relatively new option "Create Goto Index on Launch".  I would also be ok with moving all of the other binary options (the last four on the current menu).

**Comment by ZanderBrown** [comment]
I think it would be better for the original options to remain where they are but we could add them to the dialog

tl;dr copy not move

**Comment by craigwi** [comment]
See commit comments for the few tweaks I made.  Thanks!
